### PR TITLE
Milestone4 + Milestone 5

### PR DESCRIPTION
**Pull request by Tahirbhalli**
In this Milestone I add:

Complete Project
Run Locally:
> npm install
> npm start

Note: Currently Project is deployed of feature branch which is practice project has a different design.

### CODE REVIEW

File: src/component/Button.js [hunk 1]
```diff
@@ -0,0 +1,26 @@
+/* eslint-disable react/prop-types */
+/* eslint-disable react/destructuring-assignment */
+/* eslint-disable react/button-has-type */
```
Comment (suretrust) [inline src/component/Button.js]: I see no reason why you should disable these rules, you just need to understand them and write your code accordingly. 

- [x] react/prop-types => If the `propType` does not have `isRequired`, ensure you add `defaultProps` value.

- [x] react/destructuring-assignment => destructure where applicable, that's simply es6, which is a requirement for this project.

- [x] react/button-has-type => add `type` attribute, example: `type="submit"`.

Read more about `PropTypes` [here](https://reactjs.org/docs/typechecking-with-proptypes.html).

File: src/component/Button.js [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+import React from 'react';
+import PropTypes from 'prop-types';
+
+export default function Button(props) {
+  const { handleclick } = props;
+  if (props.wide) {
+    return (
+      // eslint-disable-next-line no-useless-concat
+      <button className={'double ' + `${props.color}`} onClick={() => handleclick(props.name)}>{props.name}</button>
```
Comment (suretrust) [inline src/component/Button.js]: - [ ] no-useless-concat => you could just use es6 entirely. 
```suggestion
      <button className={`double ${props.color}`} onClick={() => handleclick(props.name)}>{props.name}</button>
```

File: src/component/Button.js [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+    );
+  }
+  return (
+    <button className={`${props.color}`} onClick={() => handleclick(props.name)}>{props.name}</button>
+  );
+}
+Button.prototype = {
```
Comment (suretrust) [inline src/component/Button.js]:
```suggestion
Button.propTypes = {
```

File: src/component/Button.js [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+  name: PropTypes.string,
+  wide: PropTypes.bool,
+};
+
+Button.defaultProps = {
+  color: 'orange',
+};
```
Comment (suretrust) [inline src/component/Button.js]:
```suggestion
Button.defaultProps = {
  color: 'orange',
  wide: false,
};
```

File: src/component/ButtonPanel.js [hunk 5]
```diff
@@ -0,0 +1,30 @@
+/* eslint-disable react/prop-types */
```
Comment (suretrust) [inline src/component/ButtonPanel.js]: Remove the rule disable and add propTypes checking.
```suggestion
```

File: src/component/Display.js [hunk 6]
```diff
@@ -0,0 +1,16 @@
+/* eslint-disable react/prop-types */
+/* eslint-disable react/destructuring-assignment */
```
Comment (suretrust) [inline src/component/Display.js]:
```suggestion
```

File: src/component/Display.js [hunk 7]
```diff
@@ -0,0 +1,16 @@
...
+import PropTypes from 'prop-types';
+
+export default function Display(props) {
+  return (
+    <input placeholder="enter the number" value={props.result} />
+  );
+}
+Display.prototype = {
+  result: PropTypes.string,
+};
```
Comment (suretrust) [inline src/component/Display.js]:
```suggestion
Display.propTypes = {
  result: PropTypes.string,
};
```

**Review by Forison — CHANGES_REQUESTED** [review]
## Status:: Changes Required❌

# 🙋‍♂️ @Tahirbhalli,

#### You have a great job so far, but your project is very different from what the requirement ought to yield after following them🤗.

* [x] Kindly check milestone 3 for UI designs [check this out](https://microverse.pathwright.com/library/fast-track-curriculum/69047/path/step/44896082/).

* [x] The following basic functionalities are missing ie ```%, AC, -/+```.

* [x] Disabling linters will let you miss out on some important practices that you should know.

**[OPTIONAL]**

* [ ] It is better to have semantic Branch names, it would have been better to name this branch milestone 5.

## That is it kindly check on these requirements and submit a re-review🤗.

#### Happy coding👨‍💻

Addo Forison

File: src/logic/operate.js [hunk 8]
```diff
@@ -0,0 +1,27 @@
...
+      return First.minus(Second);
+    case 'x':
+      return First.times(Second);
+    case '/':
+      if (Second === 0) {
+        return 'undefined';
+      }
+      return First.div(Second);
+    case '%':
+      return First % Second;
```
Comment (suretrust) [inline src/logic/operate.js]: `%` should be treated as perentage, `59% === 59/100 === 0.59`.

File: src/logic/operate.js [hunk 9]
```diff
@@ -0,0 +1,27 @@
...
+    case '+':
+      return First.plus(Second);
+    case '-':
+      return First.minus(Second);
+    case 'x':
+      return First.times(Second);
+    case '/':
+      if (Second === 0) {
+        return 'undefined';
+      }
```
Comment (suretrust) [inline src/logic/operate.js]: Let's handle this better, it should give some explanatory error, like `Can't divide by 0`, `undefined` is kind of vague.

**Review by suretrust — CHANGES_REQUESTED** [review]
Hi @Tahirbhalli, 👋 

### Great job thus far! 🙌 Please see my comments. 👇 

- [ ] All the `eslint` rules that you disabled are needed, let's remove all and follow best and modern practices.

**KIndly submit another CRR once these issues have been solved.** _Happy coding!_ 🎉

**Review by shloch — APPROVED** [review]
## Hello @Tahirbhalli 
accomplished

Great job with bringing the modifications

Keep up with the good work
Happy coding

File: src/logic/operate.js [hunk 10]
```diff
@@ -0,0 +1,27 @@
...
+    case '+':
+      return First.plus(Second);
+    case '-':
+      return First.minus(Second);
+    case 'x':
+      return First.times(Second);
+    case '/':
+      if (Second === 0) {
+        return 'cannot divide by zero';
+      }
```
Comment (shloch) [inline src/logic/operate.js]: i see the code for managing this error, but it's not working on the design, debug this :)

File: src/component/Display.js [hunk 11]
```diff
@@ -0,0 +1,15 @@
...
+  return (
+    <input placeholder="enter the number" value={result} />
+  );
+}
+Display.propTypes = {
+  result: PropTypes.string,
+};
+Display.defaultProps = {
+  result: '0',
+};
```
Comment (shloch) [inline src/component/Display.js:15]: make this `ZERO` appear on startup screen

File: src/App.css [hunk 12]
```diff
@@ -1,38 +1,39 @@
-.App {
-  text-align: center;
+#root {
+  width: 700px;
 }
```
Comment (shloch) [inline src/App.css]: You can center the calculator
```
margin: auto;
```

**Review by shloch — CHANGES_REQUESTED** [review]
I reviewed for Milestone 4 above, then I claimed same project for Milestone 5....so this is a continuation

## Milestone 5 review

calc

Works very well, 
- except for division by zero
- on milestone 2, the default display on `Display` component, needed to have `0` on it, your calculator starts with a blank screen, debug this too

### [optional changes]

- since there's enough space on the buttons, can you increase the size of the button text ?? (for more visibility) 
- Can you center the whole calculator on screen? (for symmetry )

Please debug these little things, and file for last review hopefully

Happy coding

**Review by fatymahmed — APPROVED** [review]
## Hi @Tahirbhalli    
### Good job making the requested changes🎉🎉
## Status: Approved ✔️
_For future projects, use a different PR for each milestone_

**Pull request merged**